src/homeWork16/FileNavigator.java

```diff
@@ -0,0 +1,62 @@
+package homeWork16;
+
+import java.util.*;
+
+public class FileNavigator {
+
+    private Map<String, List<FileData>> fileMap = new HashMap<>();
+
+    public void add(String path, FileData fileData) {
+        List<FileData> fileDataList = new ArrayList<>();
+
+        if (isPathMatch(path, fileData)) {
+            if (fileMap.containsKey(path)) {
+                fileDataList = fileMap.get(path);
+                fileDataList.add(fileData);
+            } else {
+                fileDataList.add(fileData);
+                fileMap.put(path, fileDataList);
+            }
+        } else {
+            System.out.println("Path does not match");
```

Comment:
Do not use print statement, throw an exception instead.

Reply:
+++